### PR TITLE
Check table existence for clickhouse client autocompletion

### LINKED ISSUE #21829: Broken command line suggestions
**Describe the bug**

Command line suggestions don't work in the latest client version with older server:

```
$ clickhouse-client -h mycluster.com -d mydb -u app --ask-password

ClickHouse client version 21.3.2.5 (official build).
Connecting to database mydb at mycluster.com:9000 as user app.
Connected to ClickHouse server version 20.8.5 revision 54438.

ClickHouse server version is older than ClickHouse client. It may indicate that the server is out of date and can
be upgraded.

mycluster.com :) Cannot load data for command line suggestions: Code: 60, e.displayText() = DB::Exception:
Received from mycluster.com:9000. DB::Exception: Table system.errors doesn't exist..
(version 21.3.2.5 (official build))
```
Though it works well with client 21.2.6.1 (server is the same).

**Does it reproduce on recent release?**

It doesn't reproduce on 20.12 server.

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Check table existence for clickhouse client autocompletion

Covered-by: 01676_clickhouse_client_autocomplete
Fixes: #21829

### CODE REVIEW

File: programs/client/Suggest.cpp [hunk 1]
```diff
@@ -85,58 +184,81 @@ Suggest::Suggest()
...
+
+    /// EXISTS for each table maybe heavy, especially in debug build,
+    /// by default do it only if the version of the server does not match with the version of the client.
+    if (force_check)
+    {
+        if (force_check.value())
+        {
+            std::erase_if(completion, [&](const std::string & query)
+            {
+                return !queryTableExists(connection, timeouts, query);
```
Comment (sundy-li) [inline programs/client/Suggest.cpp:233]: It may cause too many queries for each table.

**Comment by alexey-milovidov** [comment]
I'm also in doubt - isn't it an overkill?

File: programs/client/Suggest.cpp (hunk 1 above)
Comment (azat) [reply programs/client/Suggest.cpp:233]: Agree, let's prefer your patch.

Comment (sundy-li) [reply programs/client/Suggest.cpp:233]: Just one more `show tables in system` query is ok.

**Comment by azat** [comment]
>I'm also in doubt - isn't it an overkill?

Rewritten using `SHOW TABLES FROM system`

**Comment by alexey-milovidov** [comment]
Superseded by #23040 to make it simple.

**Pull request closed without merging**